test/integration/component/test_secsr_mount.py

```diff
@@ -0,0 +1,170 @@
+        self.cleanup = []
+        self.services = {
+                            "service_offering_local": {
+                                "name": "Ultra Tiny Local Instance",
+                                "displaytext": "Ultra Tiny Local Instance",
+                                "cpunumber": 1,
+                                "cpuspeed": 100,
+                                "memory": 128,
+                                "storagetype": "local"
+                            },
+                            "vm": {
+                                "username": "root",
+                                "password": "password",
+                                "ssh_port": 22,
+                                # Hypervisor type should be same as
+                                # hypervisor type of cluster
+                                "privateport": 22,
+                                "publicport": 22,
+                                "protocol": 'TCP',
+                            },
+                         "ostype": 'CentOS 5.3 (64-bit)',
+                         "sleep": 30,
+                         "timeout": 10,
+                         }
+        
+
+    def tearDown(self):
+        try:
+            # Clean up, terminate the created templates
+            cleanup_resources(self.apiclient, self.cleanup)
+
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+        return
+
+
+    def isOnlyLocalStorageAvailable(self):
+        if not(self.zone.localstorageenabled):
+            return False
+        
+        storage_pools = StoragePool.list(
+                   self.apiclient,
+                   zoneid=self.zone.id,
+                   listall=True
+                    )
+        self.assertEqual(
+                           isinstance(storage_pools, list),
+                           True,
+                           "Check if listStoragePools returns a valid response"
+                           )
+        for storage_pool in storage_pools:
+            if storage_pool.type == u'NetworkFilesystem':
+                return False
+            
+        return True
+
+    def areDownloaded(self, id):
+        # Fetch details of templates copied from table in database
+        qresultset = self.dbclient.execute(
+            "SELECT template_id, download_state FROM cloud.template_spool_ref, cloud.vm_template where \
+            cloud.template_spool_ref.template_id=cloud.vm_template.id and cloud.vm_template.uuid='%s';" %
+            id)
+        self.logger.debug("Qresult %s" % format(qresultset))
+        if not isinstance(qresultset, list) or len(qresultset) <= 1:
+            return False, 1
+        
+        for result in qresultset:
+            self.logger.debug('Template %s download status %s' % (result[0], result[1])) 
+            if result[1] != 'DOWNLOADED':
+                return False, 1
+            
+        return True, 1
```

Comment:
For all methods using `wait_until`, the idea if you pass the status, object; here the object can be None. Since the returned object (1) is not used at all.

Reply:
method is changed